models/baseModels/Item/Item.ts

```diff
@@ -18,6 +18,34 @@ export class Item extends Doc {
   trackItem?: boolean;
   itemType?: 'Product' | 'Service';
   for?: 'Purchases' | 'Sales' | 'Both';
+  hasBatchNumber?: boolean;
+
+  async beforeSync() {
+    /*
+      * This code block is to prevent users from changing the value of Has Batch No Checkbox
+        of the items which already did transactions
+      * allowing users to change the value of Has Batch No of the items which already did
+        transactions will result in incorect SLEs 
+    */
+    const ifItemHasBatchNumber = await this.fyo.db.get(
+      'Item',
+      this.name!,
+      'hasBatchNumber'
+    );
+
+    if (this.hasBatchNumber == ifItemHasBatchNumber.hasBatchNumber) {
+      return;
+    }
+
+    const isItemExistsInSLE = await this.fyo.db.itemHasTransactions(this.name!);
```

Comment:
The above solution means the `db.itemHasTransactions` can also be safely deleted.

It shouldn't have been added in the first place as the function is essentially a `db.count` with an `item` filter.

Reply:
corrected